File: library/src/main/java/com/eazy/uibase/daynight/view/StyleableSetAppCompatTextView.java
package com.eazy.uibase.daynight.view;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.drawable.Drawable;
import android.util.Log;
import android.util.TypedValue;
import android.widget.TextView;

import androidx.appcompat.widget.AppCompatTextView;

import com.google.auto.service.AutoService;
import com.eazy.uibase.R;
import com.eazy.uibase.daynight.styleable.IStyleableSet;
import com.eazy.uibase.daynight.styleable.StyleableSet;
import com.eazy.uibase.resources.Drawables;

@AutoService(IStyleableSet.class)
public class StyleableSetAppCompatTextView extends StyleableSet<AppCompatTextView> {

    private static final String TAG = "StyleableSetTextView";

    public StyleableSetAppCompatTextView() {
        // drawables
        addStyleable(R.attr.drawableLeftCompat, (view, value) ->
            setCompoundDrawable(view, 0, value));
        addStyleable(R.attr.drawableTopCompat, (view, value) ->
            setCompoundDrawable(view, 1, value));
        addStyleable(R.attr.drawableRightCompat, (view, value) ->
            setCompoundDrawable(view, 2, value));
        addStyleable(R.attr.drawableBottomCompat, (view, value) ->
            setCompoundDrawable(view, 3, value));
        addStyleable(R.attr.drawableStartCompat, (view, value) ->
            setCompoundDrawableRelative(view, 0, value));
        addStyleable(R.attr.drawableEndCompat, (view, value) ->
            setCompoundDrawableRelative(view, 2, value));
    }

    private void setCompoundDrawable(TextView view, int index, TypedValue value) {
        Drawable[] drawables = view.getCompoundDrawables();
        drawables[index] = Drawables.getDrawable(view.getContext(), value.resourceId);
        view.setCompoundDrawablesWithIntrinsicBounds(drawables[0], drawables[1], drawables[2], drawables[3]);
    }

    private void setCompoundDrawableRelative(TextView view, int index, TypedValue value) {
        Drawable[] drawables = view.getCompoundDrawablesRelative();
        drawables[index] = Drawables.getDrawable(view.getContext(), value.resourceId);
        view.setCompoundDrawablesRelativeWithIntrinsicBounds(drawables[0], drawables[1], drawables[2], drawables[3]);
    }

}

File: library/src/main/java/com/eazy/uibase/daynight/view/StyleableSetAppCompatTextView.kt
package com.eazy.uibase.daynight.view

import android.util.TypedValue
import androidx.appcompat.content.res.AppCompatResources
import androidx.appcompat.widget.AppCompatTextView
import com.google.auto.service.AutoService
import com.eazy.uibase.R
import com.eazy.uibase.daynight.styleable.IStyleableSet
import com.eazy.uibase.daynight.styleable.StyleableSet

@AutoService(IStyleableSet::class)
class StyleableSetAppCompatTextView : StyleableSet<AppCompatTextView>() {

    override fun defStyleAttr(): Int {
        return android.R.attr.textViewStyle
    }

    init {
        addStyleable(R.attr.drawableLeftCompat) { view: AppCompatTextView, value: TypedValue ->
            setCompoundDrawable(view, 0, value)
        }
        addStyleable(R.attr.drawableTopCompat) { view: AppCompatTextView, value: TypedValue ->
            setCompoundDrawable(view, 1, value)
        }
        addStyleable(R.attr.drawableRightCompat) { view: AppCompatTextView, value: TypedValue ->
            setCompoundDrawable(view, 2, value)
        }
        addStyleable(R.attr.drawableBottomCompat) { view: AppCompatTextView, value: TypedValue ->
            setCompoundDrawable(view, 3, value)
        }
        addStyleable(R.attr.drawableStartCompat) { view: AppCompatTextView, value: TypedValue ->
            setCompoundDrawableRelative(view, 0, value)
        }
        addStyleable(R.attr.drawableEndCompat) { view: AppCompatTextView, value: TypedValue ->
            setCompoundDrawableRelative(view, 2, value)
        }
    }


    private fun setCompoundDrawable(view: AppCompatTextView, index: Int, value: TypedValue) {
        val drawables = view.compoundDrawables
        drawables[index] = AppCompatResources.getDrawable(view.context, value.resourceId)
        view.setCompoundDrawablesWithIntrinsicBounds(drawables[0], drawables[1], drawables[2], drawables[3])
    }

    private fun setCompoundDrawableRelative(view: AppCompatTextView, index: Int, value: TypedValue) {
        val drawables = view.compoundDrawablesRelative
        drawables[index] = AppCompatResources.getDrawable(view.context, value.resourceId)
        view.setCompoundDrawablesRelativeWithIntrinsicBounds(drawables[0], drawables[1], drawables[2], drawables[3])
    }
}
